File: SwissO.Droid/EditTextDialog.cs
using Android.Content;
using Android.OS;
using Android.Views;
using Android.Widget;
using System;
using Android.App;

namespace SwissO.Droid {
    class EditTextDialog : AndroidX.Fragment.App.DialogFragment {

        private bool club;
        private ProfilManager manager;

        private View v;

        public EditTextDialog(bool club, ProfilManager manager) {
            this.club = club;
            this.manager = manager;
        }

        public override Dialog OnCreateDialog(Bundle savedInstanceState) {
            AlertDialog.Builder builder = new AlertDialog.Builder(Activity);
            LayoutInflater inflater = Activity.LayoutInflater;
            v = inflater.Inflate(Resource.Layout.dialog_edittext, null);
            builder.SetView(v);

            builder.SetTitle(club ? Resource.String.club_add : Resource.String.friend_add);
            builder.SetPositiveButton("OK", new EventHandler<DialogClickEventArgs>(OkClick));
            builder.SetNegativeButton(Resource.String.cancel, new EventHandler<DialogClickEventArgs>(CancelClick));
            Cancelable = true;
            return builder.Create();
        }


        private void OkClick(object sender, DialogClickEventArgs e) {
            string vorname = ((EditText)v.FindViewById(Resource.Id.dialog_name)).Text;
            manager.NamenCallback(club, vorname);
            Dismiss();
        }

        private void CancelClick(object sender, DialogClickEventArgs e) {
            Dismiss();
        }
    }
}

File: app/src/main/java/ch/laasch/swisso/EditTextDialog.java
package ch.laasch.swisso;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public class EditTextDialog extends DialogFragment {

    private final boolean club;
    private final ProfilFragment fragment;

    private View v;

    public EditTextDialog(boolean club, ProfilFragment fragment) {
        this.club = club;
        this.fragment = fragment;
    }

    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        v = inflater.inflate(R.layout.dialog_edittext, null);
        builder.setView(v);

        builder.setTitle(club ? R.string.club_add : R.string.friend_add);
        builder.setPositiveButton(R.string.ok, (dialog, which) -> {
            EditText et = v.findViewById(R.id.dialog_name);
            String name = et.getText().toString();
            fragment.editTextDialogResult(club, name);
            dismiss();
        });
        builder.setNegativeButton(R.string.cancel, (dialog, which) -> dismiss());
        setCancelable(true);
        return builder.create();
    }
}
